tests/graph_components/validators/test_finetuning_validator.py

```diff
@@ -0,0 +1,434 @@
+    tmp_path: Path,
+    get_validation_method: Callable[..., ValidationMethodType],
+    nlu: bool,
+    core: bool,
+):
+    # NOTE: test_model/test_validate_changing_config_epochs (2nd part)
+
+    # training
+    config1 = _get_example_config()
+    importer = _create_importer_from_config(config1, tmp_path, "config1.yml")
+    validate = get_validation_method(finetuning=False, load=False, nlu=nlu, core=core)
+    validate(importer=importer, config=config1)
+
+    # Change Config - remove parts of NLU pipeline
+    config3 = copy.deepcopy(config1)
+    config3["pipeline"] = config3["pipeline"][:1]  # drop NLU (except tokenizer)
+    importer3 = _create_importer_from_config(config3, tmp_path, "config3.yml")
+
+    # finetuning
+    loaded_validate = get_validation_method(
+        finetuning=True, load=True, nlu=nlu, core=core
+    )
+
+    # does raise - doesn't matter if it's nlu/core/both
+    with pytest.raises(InvalidConfigException):
+        loaded_validate(importer=importer3, config=config3)
+
+
+@pytest.mark.parametrize("nlu, core", [(True, False), (False, True), (True, True)])
+def test_validate_after_changing_core_config(
+    tmp_path: Path,
+    get_validation_method: Callable[..., ValidationMethodType],
+    nlu: bool,
+    core: bool,
+):
+    # NOTE: *new*
+
+    # training
+    config1 = _get_example_config()
+    importer = _create_importer_from_config(config1, tmp_path, "config1.yml")
+    validate = get_validation_method(finetuning=False, load=False, nlu=nlu, core=core)
+    validate(importer=importer, config=config1)
+
+    # Change Config - remove parts of NLU pipeline
+    config3 = copy.deepcopy(config1)
+    assert len(
+        config3["policies"]
+    ), "Please update the test so that this config has some more policies."
+    config3["policies"] = config3["policies"][:1]  # drop some Policies
+    importer3 = _create_importer_from_config(config3, tmp_path, "config3.yml")
+
+    # finetuning
+    loaded_validate = get_validation_method(
+        finetuning=True, load=True, nlu=nlu, core=core
+    )
+
+    # does raise - doesn't matter if it's nlu/core/both
+    with pytest.raises(InvalidConfigException):
+        loaded_validate(importer=importer3, config=config3) @ pytest.mark.parametrize(
+            "nlu, core", [(True, False), (False, True), (True, True)]
+        )
+
+
+class DummyNluDataImporter(NluDataImporter):
+    def __init__(self, messages: List[Message]) -> None:
+        self.training_data = TrainingData(training_examples=messages)
+
+    def get_config(self) -> Dict:
+        return {}
+
+    def get_nlu_data(self, language: Optional[Text] = "en") -> TrainingData:
+        return self.training_data
+
+
+@pytest.mark.parametrize(
+    "nlu, core,key",
+    [
+        (nlu, core, key)
+        for nlu, core in [(True, False), (False, True), (True, True)]
+        for key in [INTENT, ACTION_NAME]
+    ],
+)
+def test_validate_after_removing_or_adding_intent_or_action_name(
```

Comment:
Do we have a test where we add a message that is not a new label?

Reply:
don't think that was included before - will add one